src/python_testing/TC_AVSUM_2_7.py

```diff
@@ -0,0 +1,132 @@
+        clusterAVSTR = Clusters.Objects.CameraAvStreamManagement
+        attributesAVSTR = clusterAVSTR.Attributes
+        endpoint = self.get_endpoint(default=1)
+
+        self.step(1)  # Already done, immediately go to step 2
+
+        self.step(2)
+        # Create a dummy stream ID, initially it will fail. And a new viewport, that will also fail
+        failingvideostreamID = 5
+        smallfailingviewport = Clusters.CameraAvSettingsUserLevelManagement.Structs.ViewportStruct(x1=1, y1=10, x2=1, y2=10)
+
+        self.step(3)
+        # Send a dptzsetviewport for the dodgy stream
+        await self.send_dptz_set_viewport_command(endpoint, failingvideostreamID, smallfailingviewport, expected_status=Status.NotFound)
+
+        self.step(4)
+        # Allocate the stream
+        videoStreamID = await self.video_stream_allocate_command(endpoint)
+
+        self.step(5)
+        # Send a dptzsetviewport for the correct stream but with an invalid viewport small dimension viewport
+        await self.send_dptz_set_viewport_command(endpoint, videoStreamID, smallfailingviewport, expected_status=Status.ConstraintError)
+
+        self.step(6)
+        # Send a dptzsetviewport for the correct stream but with an invalid viewport larger than the sensor call deal with dimension viewport
+        sensordimensions = await self.read_avstr_attribute_expect_success(endpoint, attributesAVSTR.VideoSensorParams)
+        largefailingviewport = Clusters.CameraAvSettingsUserLevelManagement.Structs.ViewportStruct(x1=0, y1=0,
+                                                                                                   x2=sensordimensions.sensorWidth+16,
+                                                                                                   y2=sensordimensions.sensorHeight+9)
+
+        self.step(7)
+        await self.send_dptz_set_viewport_command(endpoint, videoStreamID, largefailingviewport, expected_status=Status.ConstraintError)
+
+        # Get the current viewport
+        # Get the sensor dimensions
+        # define viewport that has x1 at sensor width - viewport width, y1 at 0
+        viewport = await self.read_avstr_attribute_expect_success(endpoint, attributesAVSTR.Viewport)
+        viewportwidth = viewport.x2 - viewport.x1
+        viewportheight = viewport.y2 - viewport.y1
+        x1 = sensordimensions.sensorWidth - viewportwidth
+
+        self.step(8)
+        passingviewport = Clusters.CameraAvSettingsUserLevelManagement.Structs.ViewportStruct(x1=x1, y1=0,
+                                                                                              x2=sensordimensions.sensorWidth,
+                                                                                              y2=viewportheight)
+        self.step(9)
+        await self.send_dptz_set_viewport_command(endpoint, videoStreamID, passingviewport)
+
+        self.step(10)
+        # Deliberately mess with the AR, ensure that the viewport setting fails.
```

Comment:
```suggestion
        # Deliberately mess with the aspect ratio, ensure that the viewport setting fails.
```
^ not important, but it took me a sec. I was thinking augmented reality

Reply:
Expanded